lib/upgradescheme.js

```diff
@@ -30,6 +31,149 @@ export class UpgradeScheme extends ExtendTruffleContract(SolidityUpgradeScheme)
     return new this(contract);
   }
 
+/*******************************************
+ * proposeController
+ */
+  async proposeController(opts={}) {
+    const defaults = {
+      /**
+       * avatar address
+       */
+      avatar: undefined
+      /**
+       *  controller address
+       */
+      , controller: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.avatar)
+    {
+      throw new Error("avatar address is not defined");
+    }
+
+    if (!options.controller)
+    {
+      throw new Error("controller address is not defined");
+    }
+
+      let tx = await this.contract.proposeUpgrade(options.avatar, options.controller);
+
+      return tx;
+  }
+
+
+/********************************************
+ * proposeUpgradingScheme
+ */
+    async proposeUpgradingScheme(opts={}) {
+
+    /**
+     * Note that explicitly supplying any property with a value of undefined will prevent the property
+     * from taking on its default value (weird behavior of default-options)
+     */
+    const defaults = {
+      /**
+       * avatar address
+       */
+      avatar: undefined
+      /**
+       *  upgrading scheme address
+       */
+      , scheme: undefined
+      /**
+       * hash of the parameters of the upgrading scheme. These must be already registered with the new scheme.
+       */
+      , schemeParametersHash: undefined
+      /**
+       * true to register organization into the scheme when the proposal is approved.
+       * If false then caller must do it manually via scheme.registerOrganization(avatarAddress).
+       * Default is true.
+       * 
+       * [This needs to be implemented]
+      , autoRegister:true
+       */
+      /**
+       * The fee that the scheme charges to register an organization in the new upgrade scheme.
+       * The controller will be asked in advance to approve this expenditure.
+       * 
+       * If the new UpgradeScheme is an Arc scheme, you may omit fee and we will
+       * obtain the values directly from the submitted scheme.
+       * Otherwise fee is required.
+       * 
+       * The fee is paid using the token given by tokenAddress.  In Wei.
+       */
+      , fee: null
+      /**
+       * address of token that will be used when paying the fee.
+       * 
+       * If the new UpgradeScheme is an Arc scheme, you may omit tokenAddress and we will
+       * obtain the values directly from the submitted scheme.
+       * Otherwise tokenAddress is required.
+       */
+      , tokenAddress: null
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.avatar)
+    {
```

Comment:
we should choose a style of where to put the {, you go back and forth in this code. I prefer putting it on the same line as the conditional

Reply:
Would be nice if lint would enforce the stuff we care about.

Reply:
definitely